### ISSUE TITLE
unitdata requires explicit flush

### ISSUE THREAD

**Issue by AdamIsrael** [juju/charm-helpers#164]
The examples for using `unitdata.kv()` should be updated to reflect that `flush()` should be explicitly called. 

It's mentioned once in the docs for [charmhelpers.core.unitdata.Storage](http://charm-helpers.readthedocs.io/en/latest/api/charmhelpers.core.unitdata.html), below all of the examples:
```
Modifications are not persisted unless flush() is called.
```

**Comment by stub42** [comment]
service manager and reactive charms handle the flush() for you, and charms should definitely *not* call flush explicitly. If they do, the reactive states for example will get persisted early rather than correctly thrown away on charm failure. So new documentation should explicitly state 'when not using the charms.reactive framework' to make it crystal clear.

**Comment by dshcherb** [comment]
@stub42 

Agreed, there is a flush in reactive which makes it a bit confusing unless you browse the source code of the framework.
